lib/iterator.js

```diff
@@ -199,6 +199,23 @@ class Iterator {
     return result + suffix;
   }
 
+  // Call a function with `this`. Will be equivalent to calling `fn(it)`.
+  //   fn <Function>
+  apply(fn) {
+    return fn(this);
+  }
+
+  // Call a function with `this` and wrap the result in an Iterator.
+  // Example:
+  // iter([1, 2])
+  //   .chainApply(([a, b]) => [a + b, a - b])
+  //   .flat()
```

Comment:
I don't like that you'll need to use `flat()` in such use case.

Reply:
I considered always calling `iter(fn(this))` but that wouldn't work nicely with `string` use-case.